src/h5web/providers/hsds/models.ts

```diff
@@ -51,17 +45,31 @@ export interface HsdsValueResponse {
   value: HDF5Value;
 }
 
-/* ---------------------------- */
-/* ----- ENTITIES & VALUES----- */
+/* ----------------- */
+/* ----- LINKS ----- */
 
-export type HsdsEntity<T extends Entity = Entity> = T & { id: HDF5Id };
+export type HsdsLink = HsdsHardLink | HsdsSymbolicLink;
+export type HsdsCollection = 'groups' | 'datasets' | 'datatypes';
 
-export type HsdsLink = HDF5HardLink | HDF5SoftLink | HsdsExternalLink;
+interface HsdsHardLink {
+  class: 'H5L_TYPE_HARD';
+  title: string;
+  collection: HsdsCollection;
+  id: HDF5Id;
+}
 
-export interface HsdsExternalLink extends Omit<HDF5ExternalLink, 'file'> {
-  h5domain: string;
+interface HsdsSymbolicLink {
+  class: 'H5L_TYPE_SOFT' | 'H5L_TYPE_EXTERNAL';
+  title: string;
+  file?: string;
+  h5path: string;
 }
```

Comment:
I've merged the soft and external links models into one. I think the name `HsdsSymbolicLink` makes sense, but note that it doesn't matter too much since it's not exported.

Reply:
Works fine for me